presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -278,9 +286,21 @@ private Type getCommonSuperTypeForDecimal(DecimalType firstType, DecimalType sec
         return createDecimalType(targetPrecision, targetScale);
     }
 
-    private Type getCommonSuperTypeForVarchar(VarcharType firstType, VarcharType secondType)
+    private Type getCommonSuperTypeForCharacterStrings(Type firstType, Type secondType)
+    {
+        int returnTypeLength = Math.max(getCharacterStringTypeLength(firstType), getCharacterStringTypeLength(secondType));
+        if (firstType instanceof VarcharType || secondType instanceof VarcharType) {
+            return createVarcharType(returnTypeLength);
+        }
+        return createCharType(returnTypeLength);
```

Comment:
This doesn't seem correct to me. Coercions shouldn't change a value. However, going from a `char(1)` to a `char(2)` actually changes the value, since now it has an extra space at the end